tangelo/linq/tests/test_simulator.py

```diff
@@ -125,6 +125,19 @@ def test_simulate_mixed_state(self):
             results[b], _ = sim.simulate(circuit_mixed)
             assert_freq_dict_almost_equal(results[b], reference_mixed, 1e-2)
 
+    def test_simulate_mixed_state_save_measures(self):
+        """ Test mid-circuit measurement (mixed-state simulation) for compatible/testable formats and backends.
+        Mixed-state do not have a statevector representation, as they are a statistical mixture of several statevectors.
+        Simulating individual shots is suitable.
+        Some simulators are NOT good at this, by design
+        """
+        results = dict()
+        for b in installed_simulator:
+            sim = get_backend(target=b, n_shots=10 ** 3)
+            results[b], _ = sim.simulate(circuit_mixed, save_mid_circuit_meas=True)
+            print(b)
+            assert_freq_dict_almost_equal(results[b], reference_mixed, 8e-2)
+
```

Comment:
A check that `sim.mid_circuit_meas_freqs` are all the same should be present. Also a check of `self.all_frequencies`

Reply:
Done